vllm/v1/utils.py

```diff
@@ -113,11 +113,16 @@ def __init__(
         pin_memory: bool,
         with_numpy: bool = True,
     ) -> None:
-        self.cpu = torch.zeros(*size,
+        self.cpu = torch.empty(*size,
                                dtype=dtype,
                                device="cpu",
                                pin_memory=pin_memory)
-        self.gpu = self.cpu.to(device)
+        self.gpu = torch.empty_like(self.cpu, device=device)
+        # Many callers (e.g. dummy warmup paths) assume the buffers start
+        # zeroed. We still allocate with empty() for performance, but
+        # sanitize here to preserve the previous semantics.
+        self.cpu.zero_()
+        self.gpu.zero_()
```

Comment:
> Our new changes skip the host-to-device memcpy that the current path performs on every allocation and instead zero-fills the GPU buffer in place with a single device-side kernel launch, which is cheaper than moving tens of megabytes of zeros across the PCIe/NVLink link.

@namanlalitnyu That makes sense, but since we need to zero out the array anyway, I think the following would have the same performance characteristics but would be a lot more readable. Or am I missing something?
```python
        self.cpu = torch.zeros(*size,
                               dtype=dtype,
                               device="cpu",
                               pin_memory=pin_memory)
        self.gpu = torch.zeros_like(self.cpu, device=device)
```